src/track/beats.h

```diff
@@ -272,6 +345,14 @@ class Beats : private std::enable_shared_from_this<Beats> {
         ThreeHalves,
     };
 
+    enum class TempoAdjustment {
+        Faster = 0b01,
+        Slower = 0b10,
+        // The following variant are used to indicate wider adjustment requests
+        MuchFaster = 0b100 | Faster,
+        MuchSlower = 0b100 | Slower,
+    };
```

Comment:
Yeah, that's a good point. The reason for doing so, as oppositie to use maybe on bit to define the direction was to allow a null type but I guess that would make sense.

Do you think the following would make more sense? The reason for that flag-style approach was to have meaningful paramater, opposition to a `bool, bool` signature.

```suggestion
    enum class TempoAdjustment {
        Faster = 0b1,
        Slower = 0b0,
        // The following variant are used to indicate wider adjustment requests
        MuchFaster = 0b10 | Faster,
        MuchSlower = 0b10 | Slower,
    };
```

Reply:
I saw that there is some more complicated bit manipulation in the code after. I think it would make sense to encapsulate that by wrapping the enum in a class and the implementing the possible operations as simple member functions. This shouldn't introduce a huge amount of boilerplate and would the code more robust and easier to understand.